agent/api/task_unix.go

```diff
@@ -40,6 +42,89 @@ const (
 	bytesPerMegabyte  = 1024 * 1024
 )
 
+// Task is the internal representation of a task in the ECS agent
+type Task struct {
```

Comment:
All of this code duplication is leading me to think that may be this is not the best way to do this. Because, adding a new field to the task now requires modifications in 2 different files. Also, 2 * code = 2 * bugs. How about something like this:
```
// task.go, no build flags
type Task struct {
 ... // all common fields
 additionalFields
}

// task_unix.go, with the negative windows platform flag
type additionalFields struct {
 // empty for now
}

// task_windows.go, with the windows build flag
type additionalFields struct {
 cpuUnboundedEnabled bool
}
```

If that sounds reasonable, we should rewrite `config.go` in a similar vein as well. I'm sure you can come with a better name for `additionalFields` :)

Reply:
I think we should try to use the same struct. The Task API from the end user perspective is going to be the same, and forking it to have different implementations internally may become confusing down the road for us. What would you think about simply ignoring unrelated / unimplemented fields on each platform? This is the pattern used in the Docker API, and it works pretty well.